src/rai/rai/communication/sound_device/api.py

```
@@ -0,0 +1,279 @@
+                samplerate=self.sample_rate,
+                channels=self.config.channels,
+                device=self.device_number,
+                dtype=self.config.dtype,
+                blocksize=window_size_samples,
+                callback=callback,
+                finished_callback=on_done,
+            )
+        except AttributeError:
+            raise SoundDeviceError(
+                f"Device {self.device_name} has not been correctly configured"
+            )
```

Ruff (0.8.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)